File: leetcode/1039-find-the-town-judge/find-the-town-judge.py
# Amir Mukhtarov, mukhtarov.amir.a@gmail.com
from typing import List


class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        trusted = [0] * n

        for (a, b) in trust:
            trusted[a - 1] -= 1
            trusted[b - 1] += 1

        for i in range(len(trusted)):
            if trusted[i] == n - 1:
                return i + 1
        return -1


if __name__ == "__main__":
    n = 3
    trust = [[1, 3], [2, 3]]
    s = Solution()
    print(s.findJudge(n, trust))

File: leetcode/1039-find-the-town-judge/find-the-town-judge.cpp
// Amir Mukhtarov, mukhtarov.amir.a@gmail.com
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <unordered_map>
#include <map>
#include <unordered_set>
#include <set>
#include <queue>
#include <stack>
#include <deque>
#include <cmath>
#include <limits>

using namespace std;

class Solution
{
public:
  int findJudge(int n, vector<vector<int>> &trust)
  {
    vector<int> trusted(n, 0);

    for (int i = 0; i < trust.size(); i++ ) {
      trusted[trust[i][0] - 1] -= 1;
      trusted[trust[i][1] - 1] += 1;
    }

    for (int i = 0; i < trusted.size(); i++ ) {
      if (trusted[i] == n - 1) {
        return i + 1;
      }
    }
    return -1;
  }
};

int main()
{

  cout << 1000 << endl;

  int n = 3;
  vector<vector<int>> trust = {{1, 3}, {2, 3}};

  Solution s = Solution();
  cout << s.findJudge(n, trust);

  return 0;
}
